torch/_weights_only_unpickler.py

```diff
@@ -207,6 +207,80 @@ def _get_allowed_globals():
     return rc
 
 
+def get_globals_in_pkl(file) -> Set[str]:
+    globals_in_checkpoint = set()
+    protocol = None
+    read = file.read
+    readline = file.readline
+    op_to_bytes_to_read = {
+        NEWOBJ[0]: 0,
+        REDUCE[0]: 0,
+        BUILD[0]: 0,
+        APPEND[0]: 0,
+        APPENDS[0]: 0,
+        SETITEM[0]: 0,
+        SETITEMS[0]: 0,
+        MARK[0]: 0,
+        TUPLE[0]: 0,
+        TUPLE1[0]: 0,
+        TUPLE2[0]: 0,
+        TUPLE3[0]: 0,
+        NONE[0]: 0,
+        NEWFALSE[0]: 0,
+        NEWTRUE[0]: 0,
+        EMPTY_TUPLE[0]: 0,
+        EMPTY_LIST[0]: 0,
+        EMPTY_DICT[0]: 0,
+        EMPTY_SET[0]: 0,
+        BINPERSID[0]: 0,
+        BININT[0]: 4,
+        BININT1[0]: 1,
+        BININT2[0]: 2,
+        BINFLOAT[0]: 8,
+        BINGET[0]: 1,
+        LONG_BINGET[0]: 4,
+        BINPUT[0]: 1,
+        LONG_BINPUT[0]: 4,
+    }
+    while True:
+        key = read(1)
+        if not key:
+            raise EOFError
+        assert isinstance(key, bytes_types)
+        if key[0] == GLOBAL[0]:
+            module = readline()[:-1].decode("utf-8")
+            name = readline()[:-1].decode("utf-8")
+            if protocol == 2:
+                if (module, name) in NAME_MAPPING:
+                    module, name = NAME_MAPPING[(module, name)]
+                elif module in IMPORT_MAPPING:
+                    module = IMPORT_MAPPING[module]
```

Comment:
This code is identical with one define in `def load`, why not refactor it into one function.
Also, just curious, is there a harm for trying to do NAME_MAPPING/IMPORT_MAPPING even if protocol version is different? (It would be a no-op for protocol 3+, isn't it?)

Reply:
will refactor!

Yep there is likely no harm trying to do NAME_MAPPING/IMPORT_MAPPING regardless of protocol